File: DemoServer/AdditionalLanguages/go/TextSearch/FTSWithStaticIndexSingleField/FTSWithStaticIndexSingleField.go
package main

//region Usings
import "github.com/ravendb/ravendb-go-client"
//endregion

var globalDocumentStore *ravendb.DocumentStore

func main() {
    createDocumentStore()
    createDatabase()
    fullTextSearchSingleField("pasta")
    globalDocumentStore.Close()
}

func createDocumentStore() (*ravendb.DocumentStore, error) {
    if globalDocumentStore != nil {
        return globalDocumentStore, nil
    }
    urls := []string{"http://localhost:8080"}
    store := ravendb.NewDocumentStore(urls, "testGO")
    err := store.Initialize()
    if err != nil {
        return nil, err
    }
    globalDocumentStore = store
    return globalDocumentStore, nil
}

func createDatabase() {
    databaseRecord := ravendb.NewDatabaseRecord()
    databaseRecord.DatabaseName = "testGO"
    createDatabaseOperation := ravendb.NewCreateDatabaseOperation(databaseRecord, 1)
    var err = globalDocumentStore.Maintenance().Server().Send(createDatabaseOperation)
    if err != nil {
        fmt.Printf("d.store.Maintenance().Server().Send(createDatabaseOperation) failed with %s\n", err)
    }
}

//region Demo
func fullTextSearchSingleField(searchTerm string) error {

        //region Step_1
        indexName := "Categories/DescriptionText"
        index := ravendb.NewIndexCreationTask(indexName)
        //endregion
        
        //region Step_2
        index.Map = `
            from category in docs.Categories 
            select new { 
                CategoryDescription = category.Description 
            }`
        //endregion
            
        //region Step_3
        index.Index("CategoryDescription", ravendb.FieldIndexingSearch)
        //endregion
        
        //region Step_4
        err := index.Execute(globalDocumentStore, nil, "")
        if err != nil {
            return err
        }
        //endregion
    
        session, err := globalDocumentStore.OpenSession("")
        if err != nil {
            return err
        }
        defer session.Close()
    
        //region Step_5
        query := session.QueryIndex(indexName)
        query = query.Where("CategoryDescription", "==", searchTerm)
    
        var categoriesWithSearchTerm []*Category
        err = query.GetResults(&categoriesWithSearchTerm)
        if err != nil {
            return err
        }
        //endregion
    
        return nil
}

type Category struct {
    ID          string
    Name        string
    Description string
}
//endregion

File: DemoServer/AdditionalLanguages/java/src/main/java/net/ravendb/demo/textSearch/fTSWithStaticIndexSingleField/FTSWithStaticIndexSingleField.java
package net.ravendb.demo.textSearch.fTSWithStaticIndexSingleField;

//region Usings
import net.ravendb.client.documents.indexes.AbstractIndexCreationTask;
import net.ravendb.client.documents.indexes.FieldIndexing;
import net.ravendb.client.documents.session.IDocumentSession;
import java.util.List;
//endregion
import net.ravendb.demo.common.DocumentStoreHolder;
import net.ravendb.demo.common.models.Category;

public class FTSWithStaticIndexSingleField {

    //region Demo
    //region Step_1
    public static class Categories_DescriptionText extends AbstractIndexCreationTask {
    //endregion

        public Categories_DescriptionText() {
            //region Step_2
            map = "docs.Categories.Select(category => new { " +
                "    CategoryDescription = category.Description " +
                "})";
            //endregion

            //region Step_3
            index("CategoryDescription", FieldIndexing.SEARCH);
            //endregion
        }
    }
    //endregion

    public List<Category> run(RunParams runParams) {
        String searchTerm = runParams.getSearchTerm();

        //region Demo
        List<Category> categoriesWithSearchTerm;

        try (IDocumentSession session = DocumentStoreHolder.store.openSession()) {
            //region Step_4
            categoriesWithSearchTerm = session.query(Category.class, Categories_DescriptionText.class)
                .whereEquals("CategoryDescription", searchTerm)
                .toList();
            //endregion
        }
        //endregion

        return categoriesWithSearchTerm;
    }

    public static class RunParams {
        private String searchTerm;

        public String getSearchTerm() {
            return searchTerm;
        }

        public void setSearchTerm(String searchTerm) {
            this.searchTerm = searchTerm;
        }
    }
}
